src/libraries/System.Private.CoreLib/src/System/Byte.cs

```diff
@@ -272,5 +279,704 @@ object IConvertible.ToType(Type type, IFormatProvider? provider)
+        //     => checked((byte)(left / right));
+
+        //
+        // IEqualityOperators
+        //
+
+        [RequiresPreviewFeatures]
+        static bool IEqualityOperators<byte, byte>.operator ==(byte left, byte right)
+            => left == right;
+
+        [RequiresPreviewFeatures]
+        static bool IEqualityOperators<byte, byte>.operator !=(byte left, byte right)
+            => left != right;
+
+        //
+        // IIncrementOperators
+        //
+
+        [RequiresPreviewFeatures]
+        static byte IIncrementOperators<byte>.operator ++(byte value)
+            => value++;
+
+        // [RequiresPreviewFeatures]
+        // static checked byte IIncrementOperators<byte>.operator ++(byte value)
+        //     => checked(value++);
+
+        //
+        // IMinMaxValue
+        //
+
+        [RequiresPreviewFeatures]
+        static byte IMinMaxValue<byte>.MinValue => MinValue;
+
+        [RequiresPreviewFeatures]
+        static byte IMinMaxValue<byte>.MaxValue => MaxValue;
+
+        //
+        // IModulusOperators
+        //
+
+        [RequiresPreviewFeatures]
+        static byte IModulusOperators<byte, byte, byte>.operator %(byte left, byte right)
+            => (byte)(left % right);
+
+        // [RequiresPreviewFeatures]
+        // static checked byte IModulusOperators<byte, byte, byte>.operator %(byte left, byte right)
+        //     => checked((byte)(left % right));
+
+        //
+        // IMultiplicativeIdentity
+        //
+
+        [RequiresPreviewFeatures]
+        static byte IMultiplicativeIdentity<byte, byte>.MultiplicativeIdentity => 1;
+
+        //
+        // IMultiplyOperators
+        //
+
+        [RequiresPreviewFeatures]
+        static byte IMultiplyOperators<byte, byte, byte>.operator *(byte left, byte right)
+            => (byte)(left * right);
+
+        // [RequiresPreviewFeatures]
+        // static checked byte IMultiplyOperators<byte, byte, byte>.operator *(byte left, byte right)
+        //     => checked((byte)(left * right));
+
+        //
+        // INumber
+        //
+
+        [RequiresPreviewFeatures]
+        static byte INumber<byte>.One => 1;
+
+        [RequiresPreviewFeatures]
+        static byte INumber<byte>.Zero => 0;
+
+        [RequiresPreviewFeatures]
+        static byte INumber<byte>.Abs(byte value)
+            => value;
+
+        [RequiresPreviewFeatures]
+        static byte INumber<byte>.Clamp(byte value, byte min, byte max)
+            => Math.Clamp(value, min, max);
+
+        [RequiresPreviewFeatures]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        static byte INumber<byte>.Create<TOther>(TOther value)
+        {
+            if (typeof(TOther) == typeof(byte))
+            {
+                return (byte)(object)value;
+            }
+            else if (typeof(TOther) == typeof(char))
+            {
+                return checked((byte)(char)(object)value);
```

Comment:
Why are all of these checked?

Reply:
Unchecked appears to be the CreateTruncated instead

Reply:
We should revisit these names then.  In general unchecked is the default for C#, and you have to opt in to checked.  It's strange then for the APIs to invert that by having the "Create" name be checked and having to append a suffix on it to effectively opt-in to unchecked.

Reply:
`unchecked` is the default for C# yes, unless you are dealing with literals or many other forms of creation.

I believe it will ultimately be a bit of failure for a user to write `T x = T.Create(<TooBigLiteral>)` and for it to not fail by default. This is why the compiler emits CS0221 (`Constant value '###' cannot be converted to a 'T' (use 'unchecked' syntax to override`) for such a scenario

Reply:
It certainly threw me off, that normal Create was checked. Most important is both are available. But agree naming seems a bit off then.

Reply:
> I believe it will ultimately be a pit of failure for a user to write T x = T.Create(<TooBigLiteral>) and for it to not fail by default.

I similarly think it'll be a pit of failure for a user to write `T x = T.Create(variable)` and have that use checked arithmetic and throw overflow exceptions and the like.  We can't emulate the language's compile-time validation, but we can map to its run-time defaults.